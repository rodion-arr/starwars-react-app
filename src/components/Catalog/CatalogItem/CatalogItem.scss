@use "../../../assets/styles/breakpoints";

.catalog-item {
  border-radius: var(--global-border-radius);
  margin: 10px 0;

  &:hover {
    box-shadow: rgba(149, 157, 165, 0.2) 0 8px 24px;
  }

  &__link {
    text-decoration: none;
    color: inherit;
  }

  &__img-wrap {
    text-align: center;
    border-bottom: 2px solid var(--img-border-color);
    border-radius: var(--global-border-radius) var(--global-border-radius) 0 0;
    box-shadow: rgba(0, 0, 0, 0.1) 0 1px 3px 0, rgba(0, 0, 0, 0.06) 0 1px 2px 0;
    background-color: rgba(255, 255, 255, 0.5);
    padding-top: 2px;
    height: 207px;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &__img {
    height: 200px;
    max-width: 340px;

    &.img-placeholder {
      max-height: 100px;
    }
  }

  @include breakpoints.for-tablet {
    min-width: 340px;
    margin: 10px;
  }
}

.catalog-item-info {
  padding: var(--global-padding);
  background-color: #1e1d1d;
  border-radius: 0 0 var(--global-border-radius) var(--global-border-radius);

  &__name {
    font-weight: bold;
    text-transform: uppercase;
  }
}
