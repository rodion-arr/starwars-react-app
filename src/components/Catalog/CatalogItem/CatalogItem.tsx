import './CatalogItem.scss';
import React from "react";
import {Link} from "react-router-dom";
import {ImagesService} from "../../../services/images.service";

interface Props {
  img: string;
  title: string;
  url: string
}

export const CatalogItem = ({img, title, url}: Props): JSX.Element => {
  return (
    <div className="catalog-item">
      <Link to={url} className="catalog-item__link">
        <div className="catalog-item__img-wrap">
          <img
            className="catalog-item__img"
            src={img} alt={title}
            loading="lazy"
            onError={ImagesService.imageErrorHandler}
          />
        </div>
        <div className="catalog-item__info catalog-item-info">
          <div className="catalog-item-info__name">{title}</div>
        </div>
      </Link>
    </div>
  );
};
