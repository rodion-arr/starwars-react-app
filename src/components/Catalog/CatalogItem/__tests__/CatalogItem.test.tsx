import {CatalogItem} from "../CatalogItem";
import {render} from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom"

describe('CatalogItem', () => {
  it('should render correctly', function () {
    const {container} = render(<Router><CatalogItem img="test-image" title="test title" url="/test" /></Router>);

    expect(container).toMatchSnapshot();
  });
})
