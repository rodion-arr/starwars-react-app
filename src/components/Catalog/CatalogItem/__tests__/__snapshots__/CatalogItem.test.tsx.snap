// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogItem should render correctly 1`] = `
<div>
  <div
    class="catalog-item"
  >
    <a
      class="catalog-item__link"
      href="/test"
    >
      <div
        class="catalog-item__img-wrap"
      >
        <img
          alt="test title"
          class="catalog-item__img"
          loading="lazy"
          src="test-image"
        />
      </div>
      <div
        class="catalog-item__info catalog-item-info"
      >
        <div
          class="catalog-item-info__name"
        >
          test title
        </div>
      </div>
    </a>
  </div>
</div>
`;
